### ISSUE TITLE
Build script failing on an Apple Silicon

### ISSUE THREAD

**Issue by capraMambrica** [PixarAnimationStudios/USD#2854]
### Description of Issue
So far as I can tell, I have satisfied all the dependancies, and I can build most of this, but each time I get stuck at the TBB process. The last message mentions "intel64", and I tried editing the build script to have "arm64" as a guess, but that didn't help. Do I need to change something for this to work on my Silicon Mac? I want to get familiar with the USDView application.  

STATUS: Installing TBB...
INFO: Downloading https://github.com/oneapi-src/oneTBB/archive/refs/tags/2018_U1.zip to /opt/local/USD/src/2018_U1.zip
INFO: Running "curl -# -L -o 2018_U1.zip.tmp https://github.com/oneapi-src/oneTBB/archive/refs/tags/2018_U1.zip"
    -=O#-   #      #       #
INFO: Extracting archive to /opt/local/USD/src/oneTBB-2018_U1
INFO: Patching file build/macos.clang.inc (original in build/macos.clang.inc.old)...
INFO: Running "make -j12 arch=intel64 "

### Steps to Reproduce
1. clone this repo
2. create a /opt/local folder, and set permissions to my user
3. run: python3 ~/DEV/USD_build/OpenUSD/build_scripts/build_usd.py /opt/local/USD --verbose
4. wait 

### System Information (OS, Hardware)
1. Mac Studio M2 Max 32gb
5. MacOS 14.1.2 (23B92)

### Package Versions
open USD 23.11
Python 3.7.9

### Build Flags
> python3 ~/DEV/USD_build/OpenUSD/build_scripts/build_usd.py /opt/local/USD --verbose

**Comment by BigRoy** [comment]
Could you share your full build log?

**Comment by jesschimein** [comment]
Filed as internal issue #USD-9065

**Comment by capraMambrica** [comment]
Of course. I have left it for 20-30 minutes at this point, and I see no signs of life in my Activity Monitor. 

[cleaned_build_log.txt](https://github.com/PixarAnimationStudios/OpenUSD/files/13647552/cleaned_build_log.txt)

**Comment by sunyab** [comment]
Hm. For some reason the build script is detecting the "native" architecture of your Mac as x86 instead of arm64. Are you possibly running your terminal under Rosetta for x86 emulation? What do you get if you run the `arch` command in your terminal? 

That being said, the TBB build shouldn't be hanging for you even if you were running in x86 mode, and I'm not sure what's going on. I just tried doing a similar build on my M1 and it went through without issues. I am running an older OS -- 12.6.5 -- so maybe someone else with an updated machine can try to reproduce the issue?

**Comment by gracien-app** [comment]
Yeah, it seems you are running your environment through Rosetta 2 (`x86_64` under `arm64`).

If this was your intention, you also have to make sure that other parts of your environment match. I would encourage you to try steps linked [here](https://github.com/PixarAnimationStudios/OpenUSD/issues/2559#issuecomment-1694676403), preferably without the RenderMan specific flags.

This is a wild guess, but I'd also upgrade Python from 3.7.9. I recall 3.8 and lower being somewhat problematic in some tools. `3.10` works well.

In any case, it will be helpful if you specify what architecture you are trying to build. Make sure the architecture matches what you expect, when executing the build commands (execute `arch` command in terminal to make sure).

With proper environment, you can even build universal binaries (x86_64 and arm64 binaries stitched together) on Apple Silicon.

**Comment by capraMambrica** [comment]
Thanks @sunyab and @gracien-app . 

Running `arch`, I get: `arm64`. 

I can confirm I have Rosetta installed, but I'm not intentionally using it. I will upgrade to Python 3.10 and see how that goes. 

The steps outlined in the link look worrisome to me :) Before I go further down this rabbit hole, can I just confirm something? I want to build this so I have acce to the tools (mainly the USDView application). Am I correct in thinking that there is no other release/package I should be downloading and installing?

**Comment by capraMambrica** [comment]
I have updated to Python 3.10, and updated pip and the dependancies. Now it is at least considering it an arm64:

```
INFO: Extracting archive to /opt/local/USD/src/oneTBB-2020.3
INFO: Patching file build/macos.clang.inc (original in build/macos.clang.inc.old)...
INFO: Running "make -j12 arch=arm64 "
```

**Comment by meshula** [comment]
Out of curiosity, are you sharing the /opt folder with homebrew? If yes, you may run into library conflicts and that could account for difficulties. ~ building into /opt with homebrew present is not a tested configuration, and has previously caused issues for people. Library versions in homebrew change frequently which makes troubleshooting and maintenance difficult ~ everyone is on a their own set of libraries.

**Comment by gracien-app** [comment]
> Am I correct in thinking that there is no other release/package I should be downloading and installing?

Yes @capraMambrica, you are correct. The build script automatically requests building Python-based tools which include `usdview` for stage preview purposes, `usdrecord` for CLI rendering etc.

The way you run the python build script does not override this default behaviour (with `--no-python` flag), so it's all good.

If you are sure that the build environment is native `arm64`, you have all the necessary Python dependencies and the source is based on latest `release` branch (OpenUSD v23.11, definitely builds on Apple Silicon)... I guess the library conflict is possible as mentioned above?

For that, I would go into the `/opt/local/USD` (based on your build script, basically your USD build location), and open `build/OpenUSD/CMakeCache.txt`. There, you can look up from where TBB is sourced (look for `TBB_tbb_LIBRARY` in the file). Maybe there's something interesting there.

I sympathise, your experience trying to build USD with the build script and default settings shouldn't look like this.

**Comment by capraMambrica** [comment]
Thanks @meshula and @gracien-app , I do have homebrew in that folder: `/opt/homebrew`. 

My confidence I'm on a native `arm64` environment comes from being on a Mac M2 Studio, running macOS Sonoma (14.2).

@gracien-app I have looked for `CMakeCache.txt`, but I don't see it in build folder. I have `boost` and `lib-1.2.13` in there, but no `OpenUSD`. I don't think it's even getting far enough along to create that part. And I don't see this file elsewhere on my system. 

If there is anything I can do to help, please let me know. I really want this easier for artists like myself to access, and I don't mind being a guide pig :) I'm available if you want to walk through anything on my setup, and debug.

**Comment by capraMambrica** [comment]
Found it in: `/opt/local/USD/build/zlib-1.2.13`
I don't see any mention of `TBB` in that file. I have attached it for your reference.
[CMakeCache.txt](https://github.com/PixarAnimationStudios/OpenUSD/files/13795768/CMakeCache.txt)

I do see homebrew mentioned several times in there. I do use `brew` a lot for managing my dependancies.

**Comment by meshula** [comment]
We support tbb2020.3, but homebrew has onetbb/tbb2021. I'm not sure if that is compatible or not. 2020u3 has a "not available" icon on it here: https://formulae.brew.sh/formula/tbb

We are working towards onetbb compatibility.
